src/arrayMethodJoin.js

```diff
@@ -4,8 +4,24 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let result = '';
+    const lastElement = this.length - 1;
+
+    for (let i = 0; i < lastElement; i++) {
+      if (this[i] === null || this[i] === undefined) {
```

Comment:
Here you can simplify your condition just using `!this[i]`

Reply:
But element "0" of array also `false`, and I will have a mistakes, if I use `!this[i]`.